runtime/gc_vlhgc/SchedulingDelegate.cpp

```diff
@@ -183,12 +289,31 @@ MM_SchedulingDelegate::globalGarbageCollectCompleted(MM_EnvironmentVLHGC *env, U
 void
 MM_SchedulingDelegate::partialGarbageCollectStarted(MM_EnvironmentVLHGC *env)
 {
-	Assert_MM_true(0 == _partialGcStartTime);
-
 	PORT_ACCESS_FROM_ENVIRONMENT(env);
 
+	/* Don't count the very first PGC */
+	if (0 != _partialGcStartTime) {
+		uint64_t recentPgcInterval = j9time_hires_delta(_partialGcStartTime, j9time_hires_clock(), J9PORT_TIME_DELTA_IN_MICROSECONDS);
+		_averagePgcInterval = (uintptr_t)(pgcIntervalHistoricWeight * _averagePgcInterval) + (uintptr_t)((1- pgcIntervalHistoricWeight) * recentPgcInterval);
+	}
+
 	/* Record the GC start time in order to track Partial GC times (and averages) over the course of the application lifetime */
 	_partialGcStartTime = j9time_hires_clock();
+	calculatePartialGarbageCollectOverhead(env);
+}
+
+void
+MM_SchedulingDelegate::calculatePartialGarbageCollectOverhead(MM_EnvironmentVLHGC *env) {
+
+	if ((0 == _averagePgcInterval) || (0 == _historicalPartialGCTime)) {
+		/* On the very first PGC, we can't calculate overhead */
+		return;
+	}
+	
+	double recentOverhead = (double)(_historicalPartialGCTime * 1000)/_averagePgcInterval;
+	_partialGcOverhead = MM_Math::weightedAverage(_partialGcOverhead, recentOverhead, pgcOverheadHistoricWeight);
+	
+	Trc_MM_SchedulingDelegate_calculatePartialGarbageCollectOverhead(env->getLanguageVMThread(), _partialGcOverhead, _averagePgcInterval/1000, _historicalPartialGCTime);
```

Comment:
Would you please make this single return function (by inverting logical condition)?